extension/observer/ecsobserver/fetcher.go

```diff
@@ -0,0 +1,88 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ecsobserver
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"go.uber.org/zap"
+)
+
+// ecsClient includes API required by TaskFetcher.
+type ecsClient interface {
+	ListTasksWithContext(ctx context.Context, input *ecs.ListTasksInput, opts ...request.Option) (*ecs.ListTasksOutput, error)
+	DescribeTasksWithContext(ctx context.Context, input *ecs.DescribeTasksInput, opts ...request.Option) (*ecs.DescribeTasksOutput, error)
+}
+
+type TaskFetcher struct {
+	logger  *zap.Logger
+	ecs     ecsClient
+	cluster string
+}
+
+type TaskFetcherOptions struct {
+	Logger  *zap.Logger
+	Cluster string
+	Region  string
+
+	// test overrides
+	ecs ecsClient
```

Comment:
change to allow
```suggestion
	// custom ecsClient Fetcher overrides
	ecs ecsClient
```

Reply:
There will be two overrides, I deleted the ec2 one to make the PR smaller. 

I think change the  var name to `ecsTestOverride` or `ecsTestClient` is easier compared w/ adding comment for each client.